test/unit/spec/common/cellComponents/tabs/jobStateListRow-Spec.js

```diff
@@ -332,4 +346,174 @@ define([
             itHasRowStructure();
         });
     });
+
+    describe('the job action button', () => {
+        beforeEach(async function () {
+            const completedJob = JobsData.validJobs.filter((job) => {
+                return job.status === 'completed';
+            })[0];
+
+            this.jobId = 'job action button test';
+            this.job = Object.assign({}, completedJob, { job_id: 'job action button test' });
+            this.row = document.createElement('tr');
+            this.jobStateListRowInstance = createInstance();
+
+            await this.jobStateListRowInstance.start({
+                clickAction: (e) => {
+                    e.stopPropagation();
+                    this.clickResult = 'Click detected!';
+                },
+                jobState: this.job,
+                name: 'testObject',
+                node: this.row,
+            });
+        });
+        it('should trigger the click action when clicked', function () {
+            expect(this.clickResult).toBeUndefined();
+            this.row.querySelector('button').click();
+            expect(this.clickResult).toEqual('Click detected!');
+        });
+    });
+
+    function createMutationObserver(documentElement, elementStateFunction, doThisFirst) {
+        return new Promise((resolve) => {
+            if (elementStateFunction()) {
+                resolve();
+            }
+            const observer = new MutationObserver(() => {
+                if (elementStateFunction()) {
+                    observer.disconnect();
+                    resolve();
+                }
+            });
+            observer.observe(documentElement, { attributes: true, childList: true, subtree: true });
+            doThisFirst();
+        });
+    }
+
+    describe('row selection', () => {
+        let container;
+
+        beforeEach(() => {
+            container = document.createElement('div');
+            container.innerHTML = t('table')({}, [
```

Comment:
To test this properly, I have set up a DataTables table and populated the rows with JobStateListRow instances (basically what the JobStateList widget does).